sway/desktop/render.c

```diff
@@ -423,31 +461,31 @@ static void render_titlebar(struct sway_output *output,
 	box.width = width;
 	box.height = titlebar_border_thickness;
 	scale_box(&box, output_scale);
-	render_rect(output, output_damage, &box, color);
+	render_rect(output, &clipped_output_damage, &box, color);
```

Comment:
We should no't need to clip the titlebar. Instead, just use the right dimensions when drawing it.

Reply:
I used this approach because it contains almost zero extra logic and code, and is very localized. If an error occurs it can be tracked somewhere in these few lines of code. Would there be any benefit in going all over this function and modifying code for this corner case?

Reply:
render_titlebar takes a single "width" argument. It's a bug if the caller sets the "width" argument so that we'd render outside the bounds we've set for the container. We shouldn't add extra logic and code to work around such bugs.

Did you have an observed case where this happens?

(I don't personally think we need to consider/add code for the scenario where the titlebar is *taller* than the container)

Reply:
No, I have not observed any issues with the width and I agree that in such a case, the width should be correctly specified by the caller. The only issue that the title bar clipping part of this PR is aiming to address is in stacked containers.